### PR TITLE
Setup Drizzle and Postgres

### PR DESCRIPTION
**Pull request by Nikhil-42**
Includes a docker-compose for launching a local Postgres server, if you need help with Docker let me know. 

Connects Drizzle to a postgres server running on localhost.

Creates a Users table to ensure everything is integrated properly. 

Migrations must be checked into git and two people must NEVER create new migrations at the same time (if we do then we have to coordinate merging and probably do a database reset). In a production environment, the migrations should be applied to the database only on a push to `main`. Because we are not hosting we will just have to communicate.

### CODE REVIEW

**Comment by pyukey** [comment]
Typically, I don't think it's good to hard-code environment variables like POSTGRES_PASSWORD and dbCredentials, and would instead use an .env.development file that holds all our environmental variables. However, I've never worked on a project where you locally deploy the database, so I'm not sure if these practices still apply.

**Comment by pyukey** [comment]
Also, could you comment here, for future reference, what commands you use to perform migrations?
Is it `npx drizzle-kit generate` followed by `npm run migrate`?

**Comment by Nikhil-42** [comment]
> Typically, I don't think it's good to hard-code environment variables like POSTGRES_PASSWORD and dbCredentials, and would instead use an .env.development file that holds all our environmental variables. However, I've never worked on a project where you locally deploy the database, so I'm not sure if these practices still apply.
> 
> 

Yeah, normally this would be specified in your hosting service but because we are self hosting it really doesn't matter.

I'll change it regardless.

**Comment by Nikhil-42** [comment]
@pyukey Review re-requested

File: README.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ npm run dev
 npm run dev -- --open
 ```
 
+If you change the schema of the database you will need to run the `npm run generate` command to create a new migration file. This will create a new file in the `migrations` folder that you can run with the `npm run migrate` command.
```
Comment (pyukey) [inline README.md:28]: `npx drizzle-kit generate` probably works better but this is good

**Review by pyukey — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Nikhil-42) [reply README.md:28]: I set up `npm run generate` as an alias for `npx drizzle-kit generate`. Just to keep everything documented in the project.json.

**Pull request merged**